django/db/static/db/js/main.js

```diff
@@ -0,0 +1,168 @@
+define(['jquery', 'jqueryui'], function (jQuery, jqueryui) {
+
+/**
+ * Create a download survey popup dialog that will serve a Qualtrics survey.
+ * The 'QUALTRICS_SURVEY_ID' template variable should be set in the Django view.
+ */
+jQuery(document).ready(
+    function ($) {
+        var DIALOG_TITLE_TEXT = 'Please answer a survey about your data usage:';
+        var SURVEY_HOST = 'https://hms.az1.qualtrics.com';
+        var SURVEY_URL = SURVEY_HOST + '/SE/?SID=' + QUALTRICS_SURVEY_ID;
+        var RETRY_INTERVAL = 3;
+        var SHOW_ON_INITIAL = true;
+        var COOKIE_NAME = 'hmsLINCSsurvey';
+        var EXPIRES_ON = 'Fri, 31 Dec 9999 23:59:59 GMT';
+        var COOKIE_PATTERN = new RegExp( COOKIE_NAME + "=([^;]+)", "i" );
+        var FACILITY_ID_PATTERN = new RegExp( "datasets/(\\d+)/", "i" );
+        var MAX_STORED_FACILITY_IDS = 20;
+        var $dialog = $('#modal-download-dialog');
+        
+        if (QUALTRICS_SURVEY_ID 
+            && window.location.pathname.search(/.*\/results$/) != -1) 
+        {
+            $('#search_export').find('a').click(function(e) {
+              // Firefox issue: 
+              // The download event blocks the survey iframe from loading.
+              // Prevent the default download event, and
+              // send the download to a hidden frame.
+              e.preventDefault();
+              document.getElementById('hiddenDownloadiFrame').src = e.target.href;
```

Comment:
Could you create this iframe dynamically right here instead of putting it in the HTML template? I feel that would be cleaner.